src/lib/types/stores/client-instance-store.ts

```diff
@@ -0,0 +1,24 @@
+import { Store } from './store';
+
+export interface IClientInstance extends INewClientInstance {
+    createdAt: Date;
+}
+
+export interface INewClientInstance {
+    appName: string;
+    instanceId: string;
+    sdkVersion?: string;
+    clientIp?: string;
+    lastSeen?: Date;
+}
+export interface IClientInstanceStore
+    extends Store<
+    IClientInstance,
+    Pick<INewClientInstance, 'appName' | 'instanceId'>
```

Comment:
I find it a bit hard to read when using Pick inside the generic type. Can we at least extract the type (`Pick<INewClientInstance, 'appName' | 'instanceId'>`) into a separate interface.

Reply:
Yes, I think we can :)